### PR TITLE
Add a manifest file for each ecosystem

### LINKED ISSUE #44: Add a manifest file for each ecosystem
Currently, there is no easy way to get an overview of which packages, and which versions of these packages, are in the dataset.  To make this easier, we should add a per-ecosystem text or JSON manifest file:

```text
libcraftsuperre 10.62
pycrypterexe 1.1.1 1.1.0 1.0.1 1.0.0 1.0.4 1.0.5 1.0.7 1.0.6 1.0.9 1.1.3 0.0.0
```
The CI job can automatically update the manifests as new samples are added.

### PR DESCRIPTION
**Pull request by ikretz**
This PR adds a JSON manifest file for each ecosystem that describes what packages and which versions are present in the dataset as it currently is.  It also adds to the `sync-malicious-packages` script so that these manifests are automatically updated as new samples are added going forward.

Each manifest is a JSON-encoded dictionary whose keys are the package names and whose values are sorted lists of the versions of each package present in the dataset.

Closes #44

### CODE REVIEW

File: scripts/sync-malicious-packages/sync-malicious-packages.py [hunk 1]
```diff
@@ -16,6 +19,25 @@ def parse_arguments():
...
+
+
+def update_manifest(destination, samples):
+  manifest_path = os.path.join(destination, "manifest.json")
+
+  with open(manifest_path, 'r') as f:
+    manifest = json.loads(f.read())
+
+  for sample in samples:
+    package, version = sample["package_name"], sample["package_version"]
```
Comment (sobregoso) [inline scripts/sync-malicious-packages/sync-malicious-packages.py:31]: > sample["package_name"]
Don't this this has `npm|` in it?

Comment (ikretz) [reply scripts/sync-malicious-packages/sync-malicious-packages.py:31]: Yes, thank you!  Great catch, fixing.

File: README.md [hunk 2]
```diff
@@ -16,6 +16,10 @@ Current ecosystems:
 Malicious samples are available under the **[samples/](samples/)** folder and compressed as an encrypted ZIP file with the password `infected`. The date indicated as part of the file name is the 
 discovery date, not necessarily the package publication date.
 
+The samples directory for each ecosystem contains a `manifest.json`
```
Comment (christophetd) [inline README.md:19]:
```suggestion
The `samples` directory for each ecosystem contains a `manifest.json`
```

File: scripts/sync-malicious-packages/sync-malicious-packages.py [hunk 3]
```diff
@@ -16,6 +19,28 @@ def parse_arguments():
...
+    if package.startswith("npm|"):
```
Comment (christophetd) [inline scripts/sync-malicious-packages/sync-malicious-packages.py:32]: do we need to handle Go packages similarily?

Comment (ikretz) [reply scripts/sync-malicious-packages/sync-malicious-packages.py:32]: We aren't currently monitoring the Golang ecosystem, but if that ever happens, we would need to update this line.

File: README.md (hunk 2 above)
Comment (ikretz) [reply README.md]: Accepted

File: scripts/sync-malicious-packages/sync-malicious-packages.py (hunk 3 above)
Comment (christophetd) [reply scripts/sync-malicious-packages/sync-malicious-packages.py:32]: indeed

**Review by christophetd — APPROVED** [review]

**Pull request closed without merging**